### DISCUSSION TITLE
Azure Monitor: Support to select multiple resources in the same query

### DISCUSSION THREAD

**Discussion by jackalakalaka** [grafana/grafana#41807] [Data Sources]
[A copied issue](https://github.com/grafana/azure-monitor-datasource/issues/92) from an archived Azure Monitor datasource repo:

Azure Monitor doesn’t support ResourceGroups() in a dashboard variable with “Multi-value” or "All" mode selected, and implementation of the API being able to expect more than one resource group name would be extremely useful.
```
# Example from the current query inspector:
{
  "xhrStatus": "complete",
  "request": {
    "method": "GET",
    "url": "api/datasources/proxy/2/azuremonitor/subscriptions/<guid>/resourceGroups/{rgA,rgB}/resources?api-version=2018-01-01"
  },
  "response": {
    "error": {
      "code": "InvalidResourceGroup",
      "message": "The provided resource group name '{rgA,rgB}' has these invalid characters: '{,}'. The name can only be a letter, digit, '-', '.', '(', ')' or '_'."
    }
  }
}
```

Adding this would allow for dynamic population of a dashboard with multiple resource groups' data, avoiding the need for a user to manually select a dropdown variable value to see info about more than one group. This adaptability is already possible with variables that support Multi-value mode if a user selects "Repeat by variable" under "Repeat options" whereby extended panels are automatically created.

**Comment by sunker** [comment]
Yes I agree this would be a nice feature. Think it should also be possible to do using the Azure Resource Graph api, but it's gonna require a lot of refactoring. I'll label this a feature request, and hopefully we can do this in the near future.

**Comment by Purneau** [comment]
Is it correct that using this 'multi-value' mode is currently also not possible within a resource group?
E.g. I'm trying to display the CPU usage of two machines in a single graph, but if I select more than one machine in the `VM` variable (multi-value) it fails with a 404.

Edit: to be clear, I'm using the ResourceName function for this: `ResourceNames($ResourceGroup,Microsoft.Compute/virtualMachines)`

**Comment by andresmgot** [comment]
Right, the real issue here is to be able to query multiple resources of the same type in Azure Monitor Logs. Right now, it's only supported to request one resource (or resource group or subscription) as the scope of a query. Azure support to request multiple resources as far as they are the same type (resource groups, VMs...).

Once we support that, being able to populate these values from dashboard variables should be possible.

I have updated the discussion title to reflect that. As a summary this feature can be split in two:

 - [ ] Add support for multiple resources in a Azure Monitor Logs query.
 - [ ] Verify compatibility with multi-value variables.

**Comment by ovidiubuligan** [comment]
@andresmgot what is the status of this issue?

**Reply by andresmgot** [reply]
It's in our roadmap but the development has not been started yet.

**Comment by dbg-raghulkrishna** [comment]
@andresmgot any update on this feature?

**Reply by andresmgot** [reply]
the development has started, you can subscribe to this issue for more information https://github.com/grafana/grafana/issues/55372 but we don't have an ETA yet.

**Comment by andresmgot** [comment]
The development of this is completed, you can see more details at https://github.com/grafana/grafana/issues/55372. The feature will be available in 9.4.0. Thanks for your input!

**Reply by imalkevich** [reply]
@andresmgot , just to confirm, this is not part of 9.4.0-beta1 release, correct? (because I don't see this feature in this release)

**Reply by andresmgot** [reply]
Correct, this will be available in the stable 9.4.0 release (not in the beta).

**Comment by dbg-raghulkrishna** [comment]
when can we expect this release? @andresmgot

**Reply by andresmgot** [reply]
it has been released already as part of the v9.4 release series
